linstor.go

```diff
@@ -973,3 +973,28 @@ func ValidResourceName(resName string) error {
 
 	return nil
 }
+
+// LinstorifyResourceName tries to generate a valid LINSTOR name if the input currently is not.
+// If the input is already valid, it just returns this name.
+// This tries to preserve the original meaning as close as possible, but does not try extra hard.
+// Do *not* expect this function to be injective.
+// Do *not* expect this function to be stable. This means you need to save the output, the output of the function might change without notice.
+func LinstorifyResourceName(name string) (string, error) {
+	if err := ValidResourceName(name); err == nil {
+		return name, nil
+	}
+
+	re := regexp.MustCompile("[[:^word:]]")
+	newName := re.ReplaceAllLiteralString(name, "_")
+	if err := ValidResourceName(newName); err == nil {
+		return newName, err
+	}
+
+	// fulfill at least the minimal requirement
+	newName = "LS_" + newName
+	if err := ValidResourceName(newName); err == nil {
+		return newName, nil
+	}
+
```

Comment:
At this point, if would be a good idea to call out to Linstor and see if the newly created name matches any existing name. It's not really fair to mange a name and then fail during assignment because we made it the same as something that's already there. You'll need to split out the logic above this into it's own little `doLinstorifyResourseName` function to make testing easy if you do write a test for it.

Reply:
No. This information is stale per definition. One could check here, another user (whoever that is) creates the same resource. So the caller has to deal with it. The caller has to handle the failure case. Checking here does not really improve things if one can not hold a "LINSTOR cluster lock".

All in all, I have hope that we can shift that code eventually to LINSTOR itself. This allows code sharing and it solves exactly this serialization problem, as the LINSTOR controller serializes this naming. I changed my mid a bit. If we do like I proposed we have several problems if the LINSTOR people really provide such a feature:

- on the CSI I extended an interface for this (actually breaks the API)
if LINSTOR then gets the feature:
- we remove it from CSI and golinstor, we break both public APIs again.

So now I suggest we do not touch golinstor. Move that to the CSI part. Hope for LINSTOR to provide such a functionality. If that happens we remove this part from the CSI driver without breaking any compat. If not, we can still move it to golinstor later/when we have a second user.

Reply:
I agree with not touching golinstor based on your points. I'll close this PR for now.